src/transformers/models/clvp/modeling_clvp.py

```diff
@@ -0,0 +1,1445 @@
+            use_causal_attention_mask=use_causal_attention_mask,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+
+class CLVPSpeechTransformer(CLVPTransformer):
+    def __init__(self, config: CLVPSpeechConfig):
+        super().__init__(config=config)
+
+    @add_start_docstrings_to_model_forward(CLVP_SPEECH_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=BaseModelOutputWithPooling, config_class=CLVPSpeechConfig)
+    def forward(
+        self,
+        speech_ids: Optional[torch.Tensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        use_causal_attention_mask: Optional[bool] = False,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, BaseModelOutputWithPooling]:
+        r"""
+        Returns:
+
+        """
+        return super().forward(
+            input_ids=speech_ids,
+            attention_mask=attention_mask,
+            use_causal_attention_mask=use_causal_attention_mask,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+
+@add_start_docstrings(
+    """The speech model from CLVP without any head or projection on top.""",
+    CLVP_START_DOCSTRING,
+)
+class CLVPSpeechModel(CLVPPreTrainedModel):
+    config_class = CLVPSpeechConfig
+    main_input_name = "speech_ids"
+
+    _no_split_modules = ["CLVPEncoderLayer"]
+
+    def __init__(self, config: CLVPSpeechConfig):
+        super().__init__(config)
+        self.speech_model = CLVPSpeechTransformer(config)
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def get_input_embeddings(self) -> nn.Module:
+        return self.speech_model.embeddings.token_embedding
+
+    def set_input_embeddings(self, value):
+        self.speech_model.embeddings.token_embedding = value
+
+    @add_start_docstrings_to_model_forward(CLVP_SPEECH_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=BaseModelOutputWithPooling, config_class=CLVPSpeechConfig)
+    def forward(
+        self,
+        speech_ids: Optional[torch.Tensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        use_causal_attention_mask: Optional[bool] = False,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> Union[Tuple, BaseModelOutputWithPooling]:
+        r"""
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> import torch
+        >>> from transformers import CLVPSpeechModel
+
+        >>> model = CLVPSpeechModel.from_pretrained("susnato/clvp_dev")
+
+        >>> inputs = {"speech_ids": torch.tensor([[56, 8, 48, 7, 11, 23]]).long()}
```

Comment:
Where are these speech ids obtained from? We need a feature extractor for this model if dealing with `speech -> ids` in the pre-processing

Reply:
The `speech_ids` come from a `gpt2` model.